mempool/priority/mempool_test.go

```diff
@@ -829,3 +829,90 @@ func TestConcurrentCheckTxDataRace(t *testing.T) {
 	// Wait for the test goroutine to complete
 	wg.Wait()
 }
+
+func TestTxMempool_IsRejectedTx(t *testing.T) {
+	t.Run("CheckTx rejection", func(t *testing.T) {
+		txmp := setup(t, 100)
+
+		// Create a transaction that will be rejected by CheckTx (invalid format)
+		rejectedTx := []byte("invalid-tx-format")
+		txKey := types.Tx(rejectedTx).Key()
+
+		// The transaction should not be rejected initially
+		require.False(t, txmp.IsRejectedTx(txKey))
+
+		// Try to add the transaction - it should be rejected
+		err := txmp.CheckTx(rejectedTx, nil, mempool.TxInfo{})
+		require.NoError(t, err) // CheckTx method itself doesn't return error for app rejection
+
+		// The transaction should now be marked as rejected
+		require.True(t, txmp.IsRejectedTx(txKey), "Transaction rejected by CheckTx should appear in IsRejectedTx")
+	})
+
+	t.Run("PreCheck rejection", func(t *testing.T) {
+		// Setup mempool with a precheck function that rejects transactions containing "reject"
+		preCheckFn := func(tx types.Tx) error {
+			if bytes.Contains(tx, []byte("reject")) {
+				return errors.New("rejected by precheck")
+			}
+			return nil
+		}
+		txmp := setup(t, 100, WithPreCheck(preCheckFn))
+
+		rejectedTx := []byte("reject-me=test=1")
+		txKey := types.Tx(rejectedTx).Key()
+
+		// The transaction should not be rejected initially
+		require.False(t, txmp.IsRejectedTx(txKey))
+
+		// Try to add the transaction - it should be rejected by precheck
+		err := txmp.CheckTx(rejectedTx, nil, mempool.TxInfo{})
+		require.Error(t, err) // PreCheck failures return an error
+
+		// The transaction should now be marked as rejected
+		require.True(t, txmp.IsRejectedTx(txKey), "Transaction rejected by PreCheck should appear in IsRejectedTx")
+	})
+
+	t.Run("PostCheck rejection", func(t *testing.T) {
+		// Setup mempool with a postcheck function that rejects transactions with priority < 10
+		postCheckFn := func(tx types.Tx, res *abci.ResponseCheckTx) error {
+			if res.Priority < 10 {
+				return errors.New("priority too low")
+			}
+			return nil
+		}
+		txmp := setup(t, 100, WithPostCheck(postCheckFn))
+
+		// Create a transaction with low priority (will pass CheckTx but fail PostCheck)
+		rejectedTx := []byte("sender-1=test=5") // priority = 5
+		txKey := types.Tx(rejectedTx).Key()
+
+		// The transaction should not be rejected initially
+		require.False(t, txmp.IsRejectedTx(txKey))
+
+		// Try to add the transaction - it should be rejected by postcheck
+		err := txmp.CheckTx(rejectedTx, nil, mempool.TxInfo{})
+		require.NoError(t, err) // CheckTx method itself doesn't return error for postcheck failure
```

Comment:
I'm confused by this test. Why does this not expect an error but PreCheck rejection does expect an error above

```
		require.Error(t, err) // PreCheck failures return an error
```